### PR TITLE
Product design - animations

### PR DESCRIPTION
**Pull request by galdranorn**

### CODE REVIEW

File: _includes/services/product-design/product-design/moveOnScroll.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+window.addEventListener('load', () => {
+  const $item = $('[data-move="on-scroll"]')[0];
+  const $scroll = $($($item).find('[data-move="scroll"]')[0]);
+  const maxScroll = 590;
+  let scrollWidth = $(window).width() > 1200;
+
+  const movingDots = {
+    secondary: {
+      item: $($($item).find('[data-move="secondary"]')[0]),
```
Comment (siaradona) [inline _includes/services/product-design/product-design/moveOnScroll.js:9]: `.closest()` (?)

File: _includes/services/product-design/product-design/moveOnScroll.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    const scrollPercent = $($scroll).scrollTop() / maxScroll;
+
+    Object.entries(movingDots).forEach(entry => {
+      const dot = entry[1];
+      const moveUnit = dot.rotMax - dot.rotMin;
+      const newRot = dot.rotMin + moveUnit * scrollPercent;
+      const newTransform = `rotate(${newRot}deg) translateX(${dot.x}px) rotate(${newRot}deg)`;
+
+      $(dot.item).css({
+        'animation' : 'none',
```
Comment (siaradona) [inline _includes/services/product-design/product-design/moveOnScroll.js:38]: `'name': 'value',`

File: _includes/services/product-design/product-design/showOnScroll.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+window.addEventListener('load', () => {
+  const scroll = window.requestAnimationFrame || function(callback){ window.setTimeout(callback, 1000/60)};
+  const $itemsToShow = $('[data-show="on-scroll"]');
+  const animatedClass = 'animated';
+
+  const loop = () => {
+    $itemsToShow.each(function(index) {
```
Comment (siaradona) [inline _includes/services/product-design/product-design/showOnScroll.js:7]: arrow function

File: _includes/services/product-design/product-design/showOnScroll.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      const $item = $itemsToShow[index];
+
+      if (isElementInViewport($item)) {
+        $($item).addClass(animatedClass);
```
Comment (siaradona) [inline _includes/services/product-design/product-design/showOnScroll.js:11]: could you use`.toggleClass()`?

File: _includes/services/product-design/product-design/showOnScroll.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        $($item).removeClass(animatedClass);
+      }
+    });
+
+    scroll(loop);
+  }
+
+  const isElementInViewport = (item) => {
+    const rect = item.getBoundingClientRect();
+
```
Comment (siaradona) [inline _includes/services/product-design/product-design/showOnScroll.js:22]: add const with `window.innerHeight || document.documentElement.clientHeight;`

File: _sass/new-styles/modules/services/product-design/product-design/_clutch.scss [hunk 6]
```diff
@@ -197,42 +199,184 @@
...
         }
 
         &--blue {
-          transform: translate(242px, 110px);
+          transform: rotate(-60deg) translateX(275px) rotate(-60deg);
+        }
+      }
+
+      &.animated .services__figure {
+        &-circle {
```
Comment (siaradona) [inline _sass/new-styles/modules/services/product-design/product-design/_clutch.scss:294]: for all circles:
`animation: drawCircle .8s forwards;`
for `--name`:
`animation-delay: time;`

File: _sass/new-styles/modules/services/product-design/product-design/_clutch.scss [hunk 7]
```diff
@@ -197,42 +199,184 @@
...
+          &--big::after {
+            animation: drawCircle .8s .5s forwards;
+          }
+
+          &--large::after {
+            animation: drawCircle .8s .7s forwards;
+          }
+        }
+
+        &-dot {
```
Comment (siaradona) [inline _sass/new-styles/modules/services/product-design/product-design/_clutch.scss:312]: .

File: _sass/new-styles/modules/services/product-design/product-design/_design-switch.scss [hunk 8]
```diff
@@ -217,6 +258,81 @@
...
           padding-top: 100px;
         }
+
+        &__image {
+          &--illustration {
+            img {
+              opacity: 0;
+            }
+
+            &.animated.switch-animated img.product-design__switch-card__image {
```
Comment (siaradona) [inline _sass/new-styles/modules/services/product-design/product-design/_design-switch.scss:268]: .

File: _sass/new-styles/modules/services/product-design/product-design/_design-switch.scss [hunk 9]
```diff
@@ -217,6 +258,81 @@
...
+
+          &--ui-design {
+            .product-design__switch-card__image {
+              &--first, &--second, &--third, &--fourth {
+                opacity: 0;
+                transform: translateY(-25px);
+              }
+            }
+
+            &.animated.switch-animated .product-design__switch-card__image {
```
Comment (siaradona) [inline _sass/new-styles/modules/services/product-design/product-design/_design-switch.scss:291]: .

File: _sass/new-styles/modules/services/product-design/product-design/_design-switch.scss [hunk 10]
```diff
@@ -217,6 +258,81 @@
...
+          }
+
+          &--prototyping {
+            .product-design__switch-card__image {
+              &--first, &--second, &--third, &--fourth {
+                opacity: 0;
+              }
+            }
+
+            &.animated.switch-animated .product-design__switch-card__image {
```
Comment (siaradona) [inline _sass/new-styles/modules/services/product-design/product-design/_design-switch.scss:317]: .

File: _sass/new-styles/modules/services/product-design/product-design/_design-switch.scss [hunk 11]
```diff
@@ -217,6 +258,81 @@
...
+
+              &--fourth {
+                animation: rain .4s 1s forwards;
+              }
+            }
+          }
+
+          &--prototyping {
+            .product-design__switch-card__image {
+              &--first, &--second, &--third, &--fourth {
```
Comment (siaradona) [inline _sass/new-styles/modules/services/product-design/product-design/_design-switch.scss:312]: add to elements `class-name class-name--modifier` --> .class-name { opacity: 0 };

File: _sass/new-styles/modules/services/product-design/product-design/_design-switch.scss [hunk 12]
```diff
@@ -217,6 +258,81 @@
...
+
+              &--third {
+                animation: show .5s .6s forwards;
+              }
+            }
+          }
+
+          &--ui-design {
+            .product-design__switch-card__image {
+              &--first, &--second, &--third, &--fourth {
```
Comment (siaradona) [inline _sass/new-styles/modules/services/product-design/product-design/_design-switch.scss:285]: .

File: _sass/new-styles/modules/services/product-design/product-design/_hero.scss [hunk 13]
```diff
@@ -115,14 +115,48 @@
...
+      }
+    }
+
+    &--first, &--second, &--third, &--fourth {
+      &.animated {
+        transform: none;
+      }
+    }
+
+    &--first.animated {
```
Comment (siaradona) [inline _sass/new-styles/modules/services/product-design/product-design/_hero.scss:145]: .

**Review by siaradona — APPROVED** [review]

File: _includes/services/product-design/product-design/moveOnScroll.js (hunk 1 above)
Comment (galdranorn) [reply _includes/services/product-design/product-design/moveOnScroll.js:9]: `The closest() method traverses the Element and its parents (heading toward the document root) until it finds a node that matches the provided selector string.` - we're looking for children here

File: _includes/services/product-design/product-design/showOnScroll.js (hunk 4 above)
Comment (galdranorn) [reply _includes/services/product-design/product-design/showOnScroll.js:11]: after discussion not implemented

File: _sass/new-styles/modules/services/product-design/product-design/_clutch.scss (hunk 7 above)
Comment (galdranorn) [reply _sass/new-styles/modules/services/product-design/product-design/_clutch.scss:312]: not implementable - different functions (`showSecondaryDot`/`showBlueDot`/`showYellowDot`)

**Review by siaradona — APPROVED** [review]

**Pull request merged**